content/productivity/pink-it/plugin.js

```diff
@@ -0,0 +1,529 @@
+let excludeMyPlanets = true;
+let excludeOwnedPlanets = false;
+let pinkItIsRunning = false;
+let butPinkIt = null;
+let butPinkItIsReady = false;
+let myPinkZone = null;
+let hlPinkZoneCounter = 0;
+const hlPinkZoneCounterMax = 200;
+let lastSelectedPlanet = null;
+let tablePinkZoneList = null;
+let intervUpdateTablePZList = null;
+let lastWorldCoords = null;
+let pzHoveringInList = null;
+
+import { getPlayerColor } from "https://cdn.skypack.dev/@darkforest_eth/procedural";
+
+function sleep(ms) {
+	return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+function planetHasNoOwner(planet) {
+	return planet.owner === "0x0000000000000000000000000000000000000000";
+}
+
+function planetIsMine(planet) {
+	return planet.owner === df.account;
+}
+
+function createToggleButton(text, onClick, onOrOff=false) {
+	const button = {};
+   	button.element = document.createElement('button');
+	button.element.innerText = text;
+	button.onOrOff = false;
+	button.set = function(b) {
+		button.onOrOff = b;
+		if (button.onOrOff) { // on
+			button.element.style.background='#CCC';
+			button.element.style.color='#000000';
+		} else { // off
+			button.element.style.background='#333333';
+			button.element.style.color='#FFF';
+		}
+	}
+	button.set(onOrOff);
+   	button.element.addEventListener('click', ()=>{
+		button.set(!button.onOrOff);
+		onClick(button.onOrOff);
+	});
+	return button;
+}
+
+function FullButtonLine(buttons) {
+	let table = document.createElement("table");
+	table.style.textAlign = "center";
+	table.style.width = "100%";
+	table.style.tableLayout = "fixed"; // to make all buttons the same width
+	for (let i=0; i < buttons.length; ++i) {
+		let b = buttons[i];
+		let td = document.createElement("td");
+		let ele = b.element ? b.element : b;
+		ele.style.border =  "1px solid white";
+		ele.style.width = "100%";
+		td.style.width = "100%"; // to make all buttons the same width
+		td.style.padding = "5px";
+		if (i !== 0) td.style.paddingLeft = "0px";
+		td.append(ele);
+		table.append(td);
+	}
+	return table;
+}
+
+function createTableHeader(table, strArr) {
+	var tr = document.createElement('tr');
+	for (var str of strArr) {
+		var th = document.createElement('th');
+		th.innerText = str;
+		th.style.border = "1px solid white";
+		tr.appendChild(th);
+	}
+	table.appendChild(tr);
+}
+
+function addAsTd(tr, text, width=null, color=null, center=true) {
+	var td = document.createElement('td');
+	td.style.border = "1px solid white";
+	td.innerHTML = text;
+	if (width) td.width = width+"px";
+	if (color) td.style.color = color;
+	if (center) td.style["text-align"] = "center";
+	tr.appendChild(td);
+	return td;
+}
+
+function intToStrHexColor(num) {
+	var color = num.toString(16).toUpperCase();
+	if (color.length < 2) color = "0"+color;
+	return color;
+}
+function percentToStrHexColor(percent) {
+	return intToStrHexColor(parseInt(percent*255));
+}
+function getPlanetRingRadius(planet, multiplier=1.3, min=minDrawPlanetRadius) {
```

Comment:
```suggestion
}

function getPlanetRingRadius(planet, multiplier=1.3, min=minDrawPlanetRadius) {
```

spacing between functions